src/dev/impl/DevToys/ViewModels/Tools/Generators/PasswordGenerator/PasswordGeneratorToolViewModel.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using DevToys.Api.Core;
+using DevToys.Api.Core.Settings;
+using DevToys.Api.Tools;
+using DevToys.Helpers;
+using DevToys.UI.Controls;
+using DevToys.ViewModels.Tools.GuidGenerator;
+using DevToys.Views.Tools.PasswordGenerator;
+using Microsoft.Toolkit.Mvvm.ComponentModel;
+using Microsoft.Toolkit.Mvvm.Input;
+
+namespace DevToys.ViewModels.Tools.Generators.PasswordGenerator
+{
+    [Export(typeof(PasswordGeneratorToolViewModel))]
+    public sealed class PasswordGeneratorToolViewModel : ObservableRecipient, IToolViewModel
+    {
+        /// <summary>
+        /// Whether the password should include uppercase characters.
+        /// </summary>
+        private static readonly SettingDefinition<bool> Uppercase
+            = new(
+                name: $"{nameof(PasswordGeneratorToolViewModel)}.{nameof(Uppercase)}",
+                isRoaming: true,
+                defaultValue: true);
+
+        /// <summary>
+        /// Whether the password should include lowercase characters.
+        /// </summary>
+        private static readonly SettingDefinition<bool> Lowercase
+            = new(
+                name: $"{nameof(PasswordGeneratorToolViewModel)}.{nameof(Lowercase)}",
+                isRoaming: true,
+                defaultValue: true);
+
+        /// <summary>
+        /// Whether the password should include numbers.
+        /// </summary>
+        private static readonly SettingDefinition<bool> Numbers
```

Comment:
Is it really used?

Reply:
I think I originally meant to add the numbers, but then realize the feature request didn't have anything about whether or not to include an option of include/exclude numbers. Would we like to do so? Or should we omit them  from the options?

Reply:
So, currently we have the possibility to generate password _**without**_:
- lowercase letters
- uppercase letters
- symbols
If we also remove numbers... well there's not much to generate then 😅

But I do see the value for generating a password without numbers too.
In term of UX, I'm quite undecided at the moment, but in short, assuming we have 4 options
- lowercase letters
- uppercase letters
- symbols
- numbers
we should enforce at least 1 option, or display a message asking the user to enable at least 1 option in order to generate a password.

These options could be displayed as:
- Toggle check box (like today)
- List with multiple selection
- List of check box
- else? (I can't think of anything else)

What do you think?

Reply:
It would be very easy to display a message in the output text if no options are selected. The other options would require more work, but also would be acceptable.

Reply:
Could we give a try at the displaying a message?
Perhaps you can use an `InfoBar` like here: https://github.com/veler/DevToys/blob/2e99e04d6f363a5bddf044afc5545266be43246c/src/dev/impl/DevToys/Views/Tools/Text/RegEx/RegExToolPage.xaml#L118-L123 But display a `Warning` or `Information` instead of `Error`?

Another option I see (but it might be more complex too) is that when it remains only 1 option checked, we disable the checked option (the option will be checked but user won't be able to uncheck it) until another option is checked.

Reply:
I'll take a look at this. It just may be awhile since I've gotten a bit busy.